### PR TITLE
Refactored data extraction format, added export in JSON lines

### LINKED ISSUE #27: Data extractor do not split multiple selectors that do no have childs.

### PR DESCRIPTION
**Pull request by mxsnq**
See https://github.com/ispras/web-scraper-chrome-extension/pull/35#issuecomment-620416123
Closes #27

### CODE REVIEW

File: src/scripts/Selector/SelectorDocument.js [hunk 1]
```diff
@@ -27,80 +26,65 @@ export default class SelectorDocument extends Selector {
...
+		);
 
-		// return empty record if not multiple type and no elements found
-		if (this.multiple === false && elements.length === 0) {
-			var data = {};
-			data[this.id] = null;
-			dfd.resolve([data]);
-			return dfd;
+		const result = {};
+		if (this.multiple) {
```
Comment (Yatskov) [inline src/scripts/Selector/SelectorDocument.js:40]: Looks like ternary operator. But maybe 2 ternary operators is too much for one expression.

File: src/scripts/StorePouchDB.js [hunk 2]
```diff
@@ -35,8 +12,10 @@ class StoreScrapeResultWriter {
 		if (docs.length === 0) {
 			return Promise.resolve();
 		}
-		const normalizedDocs = normalizeProperties(docs);
-		return this.db.bulkDocs({ docs: normalizedDocs }).catch(err => {
+		// Extra wrapping is added since PouchDB forbids identifiers starting
+		// with underscore. We use them for our attachments and meta information.
+		// TODO Consider moving custom properties from doc to wrapper level
+		return this.db.bulkDocs({ docs: docs.map(doc => ({ data: doc })) }).catch(err => {
```
Comment (Yatskov) [inline src/scripts/StorePouchDB.js:18]: Do we need { data: doc } in parentheses?

File: src/scripts/Selector/SelectorDocument.js (hunk 1 above)
Comment (mxsnq) [reply src/scripts/Selector/SelectorDocument.js:40]: It does, but eslint disallows nested ternaries, so I split them

File: src/scripts/StorePouchDB.js (hunk 2 above)
Comment (mxsnq) [reply src/scripts/StorePouchDB.js:18]: Yes, that's how you [return object literal in arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#Returning_object_literals)

File: src/scripts/Controller.js [hunk 3]
```diff
@@ -1342,43 +1344,184 @@ export default class SitemapController {
...
-					const button_a = $('.download-button a');
-					button_a.attr('href', window.URL.createObjectURL(blob));
-					button_a.attr('download', `${sitemap._id}.csv`);
-					$('.download-button').show();
-					$('.result').hide();
-				});
-			}.bind(this)
-		);
-		Translator.translatePage();
+		$('#viewport form').bootstrapValidator({
```
[the remark this answers is not in the document]
Comment (mxsnq) [reply src/scripts/Controller.js:1354]: done
also made the form hide donwload button and alert on every change

File: src/scripts/Selector.js [hunk 4]
```diff
@@ -11,7 +12,7 @@ export default class Selector {
 	 */
 	updateData(data, features) {
 		let allowedKeys = ['id', 'type', 'selector', 'parentSelectors'];
-		//XXX no need in information from window
+		// XXX no need in information from window
```
[the remark this answers is not in the document]
Comment (mxsnq) [reply src/scripts/Selector.js]: accidentally included this while rebasing, removed

**Pull request merged**